### ISSUE TITLE
Infinite recursion when tree depth over 2

### ISSUE THREAD

**Issue by Sladav** [TylorS/snabbdom-selector#11]
I get an `Uncaught RangeError: Maximum call stack size exceeded(…)` error from the `recurse` function (https://github.com/TylorS/snabbdom-selector/blob/master/src/findMatches.ts#L44) with the basic usage example:

```js
import h from 'snabbdom/h'
import snabbdom from 'snabbdom'
import {select} from 'snabbdom-selector'

const patch = snabbdom.init([                 // Init patch function with choosen modules
  require('snabbdom/modules/class'),          // makes it easy to toggle classes
  require('snabbdom/modules/props'),          // for setting properties on DOM elements
  require('snabbdom/modules/style'),          // handles styling on elements with support for animations
  require('snabbdom/modules/eventlisteners'), // attaches event listeners
]);

const vNode = h('div', {}, [
  h('div.test', {}, [
    h('p', {key: 1}, 'Foo')
  ])
])

const matches = select('.test p', vNode)

console.log(matches)
```

I get the error any time the virtual tree has a depth of 2 or more.

This is so straight-forward I can't imagine what I'm doing wrong if anything? Is there something obvious I'm missing or is this a bug?

**Comment by TylorS** [comment]
Hello @Sladav

Thanks for the issue, I can confirm its a bug.

**Comment by jvanbruegge** [comment]
Any update on this? I just ran into this when trying to update `cyclejs-sortable`

**Comment by TylorS** [comment]
I honestly haven't spent any time on this library since it's creation. I've stopped using snabbdom myself.

PRs welcome :smile:

**Comment by jvanbruegge** [comment]
Can that be the error? If not I guess I have to continue searching
```ts
function recurse (currentNode: VNode, isParent: boolean, parentVNode?: VNode) {
    const length = currentNode.children && currentNode.children.length || 0;

    for (let i = 0; i < length; ++i) {
-      const { children } = vNode;
+      const { children } = currentNode;

      if (children && typeof children[i] !== 'string') {
        const child = children[i];
        recurse(child as VNode, false, currentNode);
      }
    }

    f(currentNode, isParent, isParent ? void 0 : parentVNode);
}
```

**Comment by TylorS** [comment]
Released in v1.2.0. Hopefully all is well now. Many thanks @jvanbruegge
